packages/core/src/utils/security/handlers/utils.ts

```diff
@@ -0,0 +1,39 @@
+import * as Either from 'fp-ts/lib/Either';
+import { AuthErr } from './types';
+
+const invalidCredsErr = Either.left({
+  name: 'Forbidden',
+  status: 403,
+  message: 'Invalid credentials used',
+  headers: {},
+});
+
+export function genRespForScheme<R>(isSchemeProper: boolean, isCredsGiven: boolean, resource: R, msg: string) {
+  const handler = [
+    {
+      test: () => isSchemeProper && isCredsGiven,
+      handle: () => Either.right(resource),
+    },
+    {
+      test: () => isSchemeProper,
+      handle: () => invalidCredsErr,
+    },
+  ].find(possibleHandler => possibleHandler.test());
+
+  return handler ? handler.handle() : Either.left(genUnauthorisedErr(msg));
+}
+
+export const genUnauthorisedErr = (msg: string): AuthErr => ({
```

Comment:
In general all that’s a `Left` should be at least an `Error` so that error handling is easier